src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNet/AspNetCoreMvc2Integration.cs

```diff
@@ -0,0 +1,324 @@
+        /// <param name="routeData">A RouteData with information about the current route.</param>
+        /// <param name="opCode">The OpCode used in the original method call.</param>
+        /// <param name="mdToken">The mdToken of the original method call.</param>
+        [InterceptMethod(
+            CallerAssembly = AspnetMvcCore,
+            TargetAssembly = AspnetMvcCore,
+            TargetType = DiagnosticSource,
+            TargetSignatureTypes = new[] { ClrNames.Void, ClrNames.Ignore, "Microsoft.AspNetCore.Mvc.Abstractions.ActionDescriptor", "Microsoft.AspNetCore.Http.HttpContext", "Microsoft.AspNetCore.Routing.RouteData" },
+            TargetMinimumVersion = Major2,
+            TargetMaximumVersion = Major2)]
+        public static void AfterAction(
+            object diagnosticSource,
+            object actionDescriptor,
+            object httpContext,
+            object routeData,
+            int opCode,
+            int mdToken)
+        {
+            if (!Tracer.Instance.Settings.IsIntegrationEnabled(IntegrationName))
+            {
+                // integration disabled
+                return;
+            }
+
+            string methodDef = $"{DiagnosticSource}.{nameof(AfterAction)}(...)";
+            var integrationContext = AspNetAmbientContext.RetrieveFromHttpContext(httpContext);
+
+            Scope aspNetCoreMvcActionScope = null;
+
+            if (integrationContext == null)
+            {
+                Log.Error($"Could not access {nameof(AspNetAmbientContext)} for {methodDef}.");
+            }
+            else
+            {
+                integrationContext.TryRetrieveScope(IntegrationName, out aspNetCoreMvcActionScope);
+            }
+
+            Action<object, object, object, object> instrumentedMethod = null;
+
+            try
+            {
+                instrumentedMethod =
+                    MethodBuilder<Action<object, object, object, object>>
+                       .Start(Assembly.GetCallingAssembly(), mdToken, opCode)
+                       .WithConcreteTypeName(DiagnosticSource)
+                       .WithParameters(diagnosticSource, actionDescriptor, httpContext, routeData)
+                       .Build();
+            }
+            catch (Exception ex)
+            {
+                // profiled app will continue working as expected without this method
+                Log.ErrorException($"Error resolving {methodDef}", ex);
+            }
+
+            try
+            {
+                // call the original method, catching and rethrowing any unhandled exceptions
+                instrumentedMethod?.Invoke(diagnosticSource, actionDescriptor, httpContext, routeData);
+            }
+            catch (Exception ex)
+            {
+                // profiled app will continue working as expected without this method
+                Log.Error($"Exception when calling {methodDef}.", ex);
```

Comment:
Same comment as above, should we throw here too?

Reply:
Same as above, this is another diagnostic method, so this is intentional.